File: Leetcode/977.Squares_of_a_Sorted_Array/C++/Squares_of_a_Sorted_Array.cpp
#include <bits/stdc++.h>
#include<vector>
#include<algorithm>
using namespace std;

class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int size=nums.size();
        for(int i=0;i<size;i++){
        	nums[i]*=nums[i];
		}
		sort(nums.begin(),nums.end());
		for(int i=0;i<size;i++){
			cout << nums[i] << " ";
		}
		return nums;
    }
};

int main(){
	vector<int> nums;	
	Solution sol;
	int n;
	while(n != -2){
		cin >> n;
		if(n != -2){
			nums.push_back(n);	
		}
	}
	sol.sortedSquares(nums);		
}

File: Leetcode/977.Squares_of_a_Sorted_Array/Python/Squares_of_a_Sorted_Array.py
# -*- coding: utf-8 -*-
"""
Created on Wed Apr  5 14:53:41 2023

@author: User
"""
from typing import List
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        for i in range(len(nums)):
            nums[i]*=nums[i]
        result=sorted(nums)
        return result
nums = [-4,-1,0,3,10]
sol=Solution()
print(sol.sortedSquares(nums))